spec/decorators/service_provider_session_decorator_spec.rb

```diff
@@ -92,4 +69,27 @@
       expect(subject.sp_name).to_not be_nil
     end
   end
+
+  describe '#sp_logo' do
+    context 'service provider has a logo' do
+      it 'returns the logo' do
+        sp_logo = 'real_logo.svg'
+        sp = build_stubbed(:service_provider, logo: sp_logo)
+
+        subject = ServiceProviderSessionDecorator.new(sp: sp, view_context: view_context)
+
+        expect(subject.sp_logo).to eq sp_logo
+      end
+    end
+
+    context 'service provider does not have a logo' do
+      it 'returns the default logo' do
+        sp = build_stubbed(:service_provider, logo: nil)
+
+        subject = ServiceProviderSessionDecorator.new(sp: sp, view_context: view_context)
+
+        expect(subject.sp_logo).to eq ServiceProviderSessionDecorator::DEFAULT_LOGO
```

Comment:
How about either changing this to test that this is the generic.svg, or add another test to make sure the DEFAULT_LOGO points to the generic.svg? As is, if we changed the constant to point to the wrong image, the test would still pass. To me, testing that the constant is called is like testing implementation, rather than testing final output.